sopel/irc/modes.py

```diff
@@ -0,0 +1,316 @@
+"""Mode management for IRC channels.
+
+The :class:`ModeParser` class is used internally by the bot to parse ``MODE``
+messages for channels. User modes are not parsed (yet), as the bot doesn't
+manage them.
+
+The goal of the parser is to return a :class:`ModeMessage` that contains the
+result: modes to add/remove with their parameter when necessary, privileges to
+add/remove for a user in a channel, and errors (ignored modes and unused
+parameters).
```

Comment:
I like referring to these as "actions" rather than a "result". Whether the "errors" count as part of that or not is up for debate; I didn't include it in the list because it's mostly irrelevant unless you happen to use a non-compliant server. Ideally, almost no one will ever see non-empty ignored modes or unused parameters lists.

```suggestion
The goal of the parser is to return a :class:`ModeMessage` containing the
actions represented by the raw message:

* channel modes added/removed (including their parameters, if any)
* privileges added/removed for user(s) in a channel

Errors (ignored modes and unused parameters) are also included, mostly for
detecting when an IRC server is not conforming to specifications.
```

Reply:
Oh I like that! It's less verbose while being more accurate. Love it!